test/IntelFPGAFunctionAttributes.ll

```diff
@@ -0,0 +1,100 @@
+;; Can be compiled using https://github.com/intel/llvm SYCL compiler from:
+;; class Foo {
+;; public:
+;;   [[intelfpga::max_global_work_dim(1),
+;;     intelfpga::max_work_group_size(1,1,1),
+;;     intelfpga::num_simd_work_items(8)]] void operator()() {}
+;; };
+;;
+;; template <typename name, typename Func>
+;; __attribute__((sycl_kernel)) void kernel(Func kernelFunc) {
+;;   kernelFunc();
+;; }
+;;
+;; void bar() {
+;;   Foo boo;
+;;   kernel<class kernel_name>(boo);
+;; }
+
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv %t.bc --spirv-ext=+SPV_INTEL_kernel_attributes -o %t.spv
+; RUN: llvm-spirv %t.spv -to-text -o %t.spt
+; RUN: FileCheck < %t.spt %s --check-prefix=CHECK-SPIRV
+
+; RUN: llvm-spirv -r %t.spv -o %t.rev.bc
+; RUN: llvm-dis < %t.rev.bc | FileCheck %s --check-prefix=CHECK-LLVM
+
+; RUN: llvm-spirv -spirv-text -r %t.spt -o %t.rev.bc
+; RUN: llvm-dis < %t.rev.bc | FileCheck %s --check-prefix=CHECK-LLVM
```

Comment:
Why do you need this?

Text representation should only be used for debug purposes, so, I don't see huge value in testing that we can correctly convert it back to LLVM IR

Reply:
+1 to Ronan's comment.

Recently I was experimenting with AP integers handling in the translator (BTW, why do they arbitrary precision integers and not arbitrary width integers?). So in theory the translator shall work with with their translation, but in practice I failed to emit both SPIR-V and LLVM IR texts for my synthetic code.